dagql/idtui/types.go

```diff
@@ -49,19 +48,6 @@ func CollectSpans(db *DB, traceID trace.TraceID) []*Span {
 		if traceID.IsValid() && span.SpanContext().TraceID() != traceID {
 			continue
 		}
-		if span.Mask && span.Parent().IsValid() {
-			masked := db.Spans[span.Parent().SpanID()]
-			if masked != nil {
-				masked.Passthrough = true
-			} else {
-				// FIXME(vito): still investigating why this happens, but in
-				// the mean time, better not to panic
```

Comment:
Oh! Does this get fixed now?

How are masked spans handled now without this block?

Reply:
tl;dr: maybe?

https://github.com/dagger/dagger/pull/7452/commits/8e0e9469a833be635857ea3fd1409fe68e64dbd5 refactors it away, so this error is gone, but there were many possible ways to trigger it.

Previously we would run the function entrypoint and require the SDK to create a span that combines `dagger.io/ui.mask` and `dagger.io/ui.passthrough` to hide the `exec /runtime` span and the "dummy" span that carries the attributes. Now we just redirect logs and traces to the function span, and mark the `exec /runtime` span internal in the core implementation (I actually still need TODO this last bit, just by adding the `[internal] ` prefix). cc @helderco @TomChv (sorry for all the pings, just connecting dots :P)

There was one occurrence of this error ages ago just after shipping OTel (back when it was still a nil panic) where it turned out a ton of data was completely missing as if it was never sent out from the engine. But I've never been able to repro (after many many attempts), and users don't seem to be reporting that, so it's possible it was just a client/engine mismatch or something weird with our infra.

Another possible cause for this is receiving child spans before parent spans, which could easily happen because that's _normally_ just how OTel works. In practice I saw this happen in our test suite because our nested `dagger` calls weren't instructed to send live telemetry. That would also manifest itself by having spans show up awkwardly at the bottom of the TUI and then "pop" in to place when their parent span arrived. That's fixed now too; we set an unambiguous env var that the `dagger` CLI detects and configures live telemetry.

So, this error log indeed doesn't exist anymore, and now I can only hope I've addressed all the issues that led to it. At least by that point we won't be possibly conflating 10 different root causes, and maybe as a result we'll get deeper analysis of what went wrong by looking at a more important symptom (i.e. missing data) rather than a cryptic log message.